test/adapters/cuda/context_tests.cpp

```diff
@@ -9,7 +9,7 @@
 #include "uur/raii.h"
 #include <thread>
 
-using cudaUrContextCreateTest = uur::urDeviceTest;
+using cudaUrContextCreateTest = uur::urDeviceTest<>;
```

Comment:
I don't love having to add `<>` everywhere.

Reply:
I don't love it either but thought it was the lesser evil. 

I don't like the "WithParam" naming scheme for structs / tests (they all have parameters even if it's just the device). This was the only way to make the naming more straightforward and avoid adding extra abstractions. Maybe I got a bit carried away with the changes :man_shrugging: 

If it makes it more palatable, I can revert the changes to test / struct names and add a `using urDeviceTest = urDeviceTestWithParam<>` inside `fixtures.h`?

Reply:
`WithParam` comes from gtest itself. The device paramter was intentially hidden to make writing tests more straight forwards. I've not dug into this enough to form a proper opinion yet, I'll come back to it.